lib/install.ps1

```diff
@@ -786,10 +796,21 @@ function find_dir_or_subdir($path, $dir) {
     return [string]::join(';', $fixed), $removed
 }
 
+function prepare_env_path ($path) {
+    if ($path -like '*$*') {
+        $path_dir = format $path @{ "dir" = $dir; "persist_dir" = $persist_dir }
+    } else {
+        $path_dir = "$dir\$($path)"
+    }
+
+    return $path_dir
+}
+
 function env_add_path($manifest, $dir, $global) {
     $manifest.env_add_path | ? { $_ } | % {
-        $path_dir = "$dir\$($_)"
-        if(!(is_in_dir $dir $path_dir)) {
+        $path_dir = prepare_env_path $_
+
+        if(!(is_in_dir $dir $path_dir) -and !(is_in_dir $persist_dir $path_dir)) {
```

Comment:
If `prepare_env_path` can be removed, it would be good to not have to change this either.